### PR TITLE
91 implements avatar for users

### PR DESCRIPTION
**Pull request by aurlic**
- Default avatar for all user when they register
- New settings page with different features : 
    - Avatar change, with size limit
    - Password change

### CODE REVIEW

File: django/src/pages/templates/extends/base.html [hunk 1]
```diff
@@ -139,6 +140,13 @@ <h6>Users Online</h6>
 		})
 	}
 
+	document.querySelectorAll('[data-route]').forEach((element) => {
```
Comment (Neffi42) [inline django/src/pages/templates/extends/base.html:143]: this was just moved elsewhere in main code you need to merge main

File: django/src/ft_auth/templates/settings.html [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+			<small>Max file size: 2MB</small>
+			<button id="upload-avatar-btn" type="submit" class="upload-btn">Upload</button>
+		</form>
+	</div>
+</div>
+
+<script>
+	function showModal() {
+		const modal = document.getElementById('avatarModal');
+		modal.style.display = 'block';
```
Comment (Neffi42) [inline django/src/ft_auth/templates/settings.html:69]: use class "hidden"

File: django/src/ft_auth/templates/settings.html [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	}
+
+	function hideModal() {
+		const modal = document.getElementById('avatarModal');
+		modal.style.display = 'none';
```
Comment (Neffi42) [inline django/src/ft_auth/templates/settings.html:74]: use class "hidden"

**Review by Neffi42 — CHANGES_REQUESTED** [review]

File: django/src/pages/templates/extends/base.html (hunk 1 above)
Comment (aurlic) [reply django/src/pages/templates/extends/base.html:143]: i did merge main

Comment (Neffi42) [reply django/src/pages/templates/extends/base.html:143]: Well this is suppose to be in router.js now

File: django/src/ft_auth/views.py [hunk 4]
```diff
@@ -68,6 +70,18 @@ def password_update(request: HttpRequest):
 	logger.info(f"Updated password of {request.user.username}.")
 	return HttpResponse(status=200)
 
+def upload_avatar(request):
+ 
+	if request.FILES.get('avatar'):
```
Comment (antoineverin) [inline django/src/ft_auth/views.py:75]: Need to protect file size here too

File: django/src/pages/templates/extends/base.html (hunk 1 above)
Comment (aurlic) [reply django/src/pages/templates/extends/base.html]: addressed in : [438d1b0](https://github.com/TheSecretOrganization/ft_transcendence/pull/247/commits/438d1b00e69532d81797f333de11c94f98e67121)

File: django/src/pages/views.py [hunk 5]
```diff
@@ -114,3 +114,10 @@ def profiles(request: HttpRequest, username: str):
 		if (target.id is game.user1.id and game.score1 > game.score2) or (target.id is game.user2.id and game.score2 > game.score1):
 			win += 1
 	return create_response(request, 'profiles.html', {'target': target, 'games': games, 'wins': win}, title=f"{target.username} Profile")
+
+@require_GET
+def settings(request: HttpRequest):
+	if not request.user.is_authenticated:
```
Comment (antoineverin) [inline django/src/pages/views.py:120]: already done in the `create_response` with the `need_authentication` to True

File: django/src/ft_auth/templates/settings.html [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+			return;
+		}
+
+		const formData = new FormData();
+
+		if (avatarInput) {
+			console.log("test")
+			formData.append('avatar', avatarInput);
+
+			fetch('/api/auth/upload-avatar/', {
```
Comment (antoineverin) [inline django/src/ft_auth/templates/settings.html:134]: could have use the `postFetch` function

**Review by antoineverin — CHANGES_REQUESTED** [review]
* wrong commit content for [c842cac](https://github.com/TheSecretOrganization/ft_transcendence/pull/247/commits/c842caca7715cb0755abacd0e85c3f5605ef1869)

File: django/src/ft_auth/templates/settings.html (hunk 3 above)
Comment (aurlic) [reply django/src/ft_auth/templates/settings.html]: adressed in : [c237b06](https://github.com/TheSecretOrganization/ft_transcendence/pull/247/commits/c237b0684f5f5df9034f7881b86370e17513c291)

File: django/src/ft_auth/templates/settings.html (hunk 2 above)
Comment (aurlic) [reply django/src/ft_auth/templates/settings.html]: adressed in : [c237b06](https://github.com/TheSecretOrganization/ft_transcendence/pull/247/commits/c237b0684f5f5df9034f7881b86370e17513c291)

File: django/src/ft_auth/views.py (hunk 4 above)
Comment (aurlic) [reply django/src/ft_auth/views.py:76]: adressed in : [d42776c](https://github.com/TheSecretOrganization/ft_transcendence/pull/247/commits/d42776cdca442763caa7de3c21eacf49a904bcf6)

File: django/src/pages/views.py (hunk 5 above)
Comment (aurlic) [reply django/src/pages/views.py]: adressed in : [d42776c](https://github.com/TheSecretOrganization/ft_transcendence/pull/247/commits/d42776cdca442763caa7de3c21eacf49a904bcf6)

File: nginx/src/css/settings.css [hunk 7]
```diff
@@ -163,7 +163,7 @@
 	justify-content: center;
 	align-items: center;
 	z-index: 1000;
-	display: none;
+	display: hidden;
```
Comment (antoineverin) [inline nginx/src/css/settings.css:166]: display `hidden` is not a valid display field

**Review by antoineverin — CHANGES_REQUESTED** [review]

File: nginx/src/css/settings.css (hunk 7 above)
Comment (aurlic) [reply nginx/src/css/settings.css]: adressed in : [6049901](https://github.com/TheSecretOrganization/ft_transcendence/pull/247/commits/6049901c8f4d3865f0c8b2a8c77bd795a1b6e960)

**Review by antoineverin — APPROVED** [review]

**Pull request merged**